### DISCUSSION TITLE
Plenti's new content panel and media organisation

### DISCUSSION THREAD

**Discussion by jamestagal** [plentico/plenti#343] [Ideas]
Hi @jimafisk 

I had some thoughts and observations on your updates to Plenti's new content panel in the CMS and how media is organised.

Instead of writing it down I just recorded a fairly short video talking through my thoughts. See below.

https://www.loom.com/share/4e5a13fde75e45938d0fa5e32bc8dc29?sid=2b109c16-bcf5-4223-9f6b-c17d0f4c8be0

Thanks again for all your work you do on Plenti. It is very much appreciated. 🙏 

Regards,
Ben

**Comment by jimafisk** [comment]
Hi @jamestagal,

Thanks for the video feedback! 
1. **Media filters:** I noticed this issue on your site as well, I hadn't actually had a site with so many media folders yet. We definitely need a way for these to take up less space. The actual images should make up 80% + of the display. I like your suggestions, I was thinking along these same lines when I originally developed the filter display (you can see some early sketches [here](https://github.com/plentico/plenti/issues/206)). 
2. **Adding Content:** I'm not sure I fully understand this, it sounds like you're describing how it currently works, but maybe I'm missing something. You can certainly have scaffolding for a content type so certain components are already added when creating new pages. This is what `_defaults.json` is used for, so just make sure that's fleshed out with the field data you want. Let me know if you're referring to something else here!

**Reply by jamestagal** [reply]
Hi @jimafisk 

Yes regarding point 2, adding content I have to explore this further. I was thinking of how you could add components on the fly and I tried adding a ```new page``` to the ```Fitlife``` theme from the CMS. The ```pages``` content has scaffolding set up for the content type.  I was able to add the components you've listed as options in the ```_schema.json``` file and could save the file but then I got an ```undefined``` field on the following page and finally no styling was applied when refreshing. The ```_defaults.json``` in this case has an array of components as follows: 

```json
{
	"components": [],
	"meta": {
		"title": "New Page",
		"desc": "Meta Description",
		"keywords": "Plenti, Svelte, Go, SSG, JAMstack, Themes"
	}
}
```
Anyway I will look again at what piece might be missing from this setup that's preventing it from working.
Regards,
Ben

**Reply by jimafisk** [reply]
> I was thinking of how you could add components on the fly

You should be able to do this. If you want your page to start with components on it, instead of starting with an empty array of components:
```
"components": [],
```

Just add the components you want to appear when creating a new page, like a `video` component for example:
```
"components": [
{
    "name": "video",
    "fields": {
        "title": "Explore Fitness Life",
        "video": {
            "url": "https://www.youtube.com/watch?v=EKUNGQ4LmH8"
        },
        "image": {
            "url": "media/video-banner.jpg",
            "alt": "Man lifting weights"
        },
        "link": {
            "title": "Watch More",
            "url": "."
        }
    }
}
],
```

**Reply by jamestagal** [reply]
ok yes I see.

However if I wanted to not have any components by default when creating a new page but add them dynamically like in the Fitlife example, would that be possible?

**Reply by jimafisk** [reply]
Yes definitely, just make sure to target the field in your `_schema.json` and set the "type" to "component". Then in the "options" you can specify whatever list of components you want available.

**Reply by jamestagal** [reply]
ok. 
So in the ```_schema.json``` would we set it to all ```fields``` like so? because each component has fields and how about the options here? keep it blank?

```json
    "fields": {
        "type": "component",
        "options": [
        ]
    }
```
